manifests/record/mx.pp

```diff
@@ -1,15 +1,36 @@
 # == Define: dns::record::mx
 #
-# Wrapper for dns::record to set an XM record.
+# Wrapper for dns::record to set an MX record.
 #
 define dns::record::mx (
   $zone,
   $data,
-  $ttl = '',
-  $preference = '0',
-  $host = $name ) {
+  $ttl        = '',
+  $preference = 10,
+  $host       = '@' ) {
 
-  $alias = "${host},MX,${zone}"
+  $alias = "${host},${zone},MX,${preference},${data}"
+
+  unless is_string($zone) and is_domain_name($zone) and !is_ip_address($zone) and $zone !~ /\.?\d+$/ {
```

Comment:
Something about line 14 isn't compatible with puppet 2.7?
https://travis-ci.org/ajjahn/puppet-dns/jobs/47328674

The validation may be a bit overkill. How about just "is_domain_name()"? Shouldn't that cover it?